src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+    private int capacity;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            this.grow();
+        }
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            this.add(value);
+        } else {
+            this.checkIndex(index);
+            this.add(arrayList[size - 1]);
+            System.arraycopy(arrayList, index, arrayList, index + 1, size - index - 2);
+            arrayList[index] = value;
+        }
```

Comment:
let's return early here and unwrap the major chunk of code
```suggestion
        if (index == size) {
            this.add(value);
            return;
        } 
            this.checkIndex(index);
            this.add(arrayList[size - 1]);
            System.arraycopy(arrayList, index, arrayList, index + 1, size - index - 2);
            arrayList[index] = value;
```